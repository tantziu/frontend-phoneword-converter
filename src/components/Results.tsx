import '../styles/Results.scss'

type ResultsProps = {
    words: string[]
}

const Results = ({words}:ResultsProps) => {
    return (
        <div className="Results">
          <h3>Words:</h3>
            <ul className="resultsList">
              {words.map(result => (
                <li key={result}>{result}</li>
              ))}
            </ul>
          </div>
    )
}

export default Results
