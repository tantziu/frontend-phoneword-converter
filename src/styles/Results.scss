.Results {
    margin: 0 20px;

    .resultsList {
        list-style: none;
        padding: 0;
        margin: 0;
        display: flex;
        flex-direction: row;
        flex-wrap: wrap;

        li {
            padding: 10px 10px;
        }
    }

}
