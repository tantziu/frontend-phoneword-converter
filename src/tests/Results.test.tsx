import renderer, { ReactTestRenderer} from 'react-test-renderer'
import ReactDOM from 'react-dom'
import Results from '../components/Results'

describe('Results', () => {
    const words= ['adg',
        'adh',
        'adi',
        'aeg',
        'aeh',
        'aei',
        'afg',
        'afh',
        'afi'
        ]
    

    it('renders nine items', () => {
        const component = renderer.create(<Results words={words} onFilter={jest.fn()}/>)

        expect(component.root.findAllByType('li').length).toEqual(9)
    })

    test('has a valid snapshot', () => {
        const component = renderer.create(
            <Results words={words} onFilter={jest.fn()}/>
        )
        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })
})
