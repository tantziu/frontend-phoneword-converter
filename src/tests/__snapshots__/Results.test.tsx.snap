// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button has a valid snapshot 1`] = `
<button
  className="Button active"
  disabled={false}
  onClick={[MockFunction]}
  type="button"
>
  Convert
</button>
`;

exports[`Results has a valid snapshot 1`] = `
<div
  className="Results"
>
  <h3>
    Words:
  </h3>
  <ul
    className="resultsList"
  >
    <li>
      adg
    </li>
    <li>
      adh
    </li>
    <li>
      adi
    </li>
    <li>
      aeg
    </li>
    <li>
      aeh
    </li>
    <li>
      aei
    </li>
    <li>
      afg
    </li>
    <li>
      afh
    </li>
    <li>
      afi
    </li>
  </ul>
</div>
`;
